templates/talmudic_relationships.html

```diff
@@ -0,0 +1,733 @@
+                var sourceNode = cleanedNodes.filter(function(n) {
+
+                        return n.key === e.source;
+                    })[0],
+
+                    targetNode = cleanedNodes.filter(function(n) {
+                        return n.key === e.target;
+                    })[0];
+
+
+                if (!sourceNode || !targetNode) {
+                    console.log("Failed to resolve link with Source: " + e.source + " and target: " + e.target);
+
+                } else {
+                    edges.push({
+                        source: sourceNode,
+                        target: targetNode,
+                        value: 1
+                    });
+                }
+            });
+
+
+            var linkedByIndex = {};
+            json.links.forEach(function(d) {
+                linkedByIndex[d.source + "," + d.target] = true;
+            });
+
+            function isConnected(a, b) {
+                return linkedByIndex[a.key + "," + b.key] || linkedByIndex[b.key + "," + a.key] || a.key == b.key;
+            }
+
+            function hasConnections(a) {
+                for (var property in linkedByIndex) {
+                    s = property.split(",");
+                    if ((s[0] == a.key || s[1] == a.key) && linkedByIndex[property]) return true;
+                }
+                return false;
+            }
+
+
+            force
+                .nodes(cleanedNodes)
+                .links(edges)
+                .start();
+
+            var link = g.selectAll(".link")
+                .data(edges)
+                .enter().append("line")
+                .attr("class", "link")
+                .style("stroke-width", nominal_stroke)
+                .style("stroke", default_link_color)
+
+
+            var node = g.selectAll(".node")
+                .data(cleanedNodes)
+                .enter().append("g")
+                .attr("class", "node")
+
+                .call(force.drag)
+
+
+            node.on("dblclick.zoom", function(d) {
+                d3.event.stopPropagation();
+                var dcx = (window.innerWidth / 2 - d.x * zoom.scale());
+                var dcy = (window.innerHeight / 2 - d.y * zoom.scale());
+                zoom.translate([dcx, dcy]);
+                g.attr("transform", "translate(" + dcx + "," + dcy + ")scale(" + zoom.scale() + ")");
+
+
+            });
+
+
+
+
+            var tocolor = "fill";
+            var towhite = "stroke";
+            if (outline) {
+                tocolor = "stroke"
+                towhite = "fill"
+            }
+
+
+
+            var circle = node.append("path")
+
+
+                .attr("d", d3.svg.symbol()
+                    .size(function(d) {
+                        return Math.PI * Math.pow(size(d.size) || nominal_base_node_size, 2);
+                    })
+                    .type(function(d) {
+                        return d.type;
+                    }))
+
+                .style(tocolor, function(d) {

```

Comment:
This could be written in a more graceful way. 
Take a look at https://github.com/d3/d3-scale/blob/master/README.md